common/http_error_response_writer.go

```diff
@@ -10,16 +10,52 @@ type HTTPError struct {
 	HTTPCode    int    `json:"-"`
 	Code        string `json:"code,omitempty"`
 	Description string `json:"description,omitempty"`
+
+	extraFields map[string]interface{}
+}
+
+func (httpError *HTTPError) AddField(key string, val interface{}) {
+	if httpError.extraFields == nil {
+		httpError.extraFields = map[string]interface{}{}
+	}
+	httpError.extraFields[key] = val
+}
+
+type KV struct {
+	K string
+	V interface{}
+}
+type wrappedError struct {
+	e      error
+	fields []KV
+}
+
+func (w wrappedError) Error() string {
+	return w.e.Error()
```

Comment:
Wouldn't you want to append the extra fields here too?

Reply:
yeah, should hav done that, willl fix